backend/tournesol/models/entity.py

```diff
@@ -25,51 +25,97 @@
 CRITERIAS = settings.CRITERIAS
 
 
-class Video(models.Model, WithFeatures, WithEmbedding):
-    """One video."""
+class Entity(models.Model, WithFeatures, WithEmbedding):
+    """
+    A generic entity that can be compared with another one.
 
+    The current model still contains fields from the previous `Video` model.
+    These fields are kept as-is for now to ease the refactor of the Tournesol
+    app, and will be replaced in the future by the `metadata` JSON field.
+    """
+    class Meta:
+        verbose_name_plural = "entities"
+
+    UID_YT_NAMESPACE = 'yt'
+
+    TYPE_VIDEO = 'video'
+    ENTITY_TYPE = [
+        (TYPE_VIDEO, 'Video'),
+    ]
+
+    uid = models.CharField(
+        unique=True,
+        max_length=144,
+        help_text="A unique identifier, build with a namespace and an external id.",
+    )
+
+    type = models.CharField(
+        max_length=32,
+        choices=ENTITY_TYPE,
+    )
+
+    metadata = models.JSONField(
+        default=dict
+    )
+
+    # TODO: specific to YouTube entities, move it somewhere else
     video_id_regex = RegexValidator(
         YOUTUBE_VIDEO_ID_REGEX, f"Video ID must match {YOUTUBE_VIDEO_ID_REGEX}"
     )
 
+    # TODO: will be replaced by the `uid` field
     video_id = models.CharField(
         max_length=20,
         unique=True,
         help_text=f"Video ID from YouTube URL, matches {YOUTUBE_VIDEO_ID_REGEX}",
         validators=[video_id_regex],
     )
```

Comment:
After this PR is deployed, this `video_id` is now redundant with `uid`.  

It does not seem that the migration has been specified yet. Should we remove it at the same time as other fields specific to videos, or can it be replaced with a `@property` now ?